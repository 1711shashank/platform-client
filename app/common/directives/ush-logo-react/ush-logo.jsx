import React from "react";
import PropTypes from "prop-types";
// import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import connectWithStore from "../../../react-transition/connectWithStore";
import * as UshLogoActions from "./ush-logo.actions";
import { getFormsFromState } from "./ush-logo.reducer";

const MyComponent = props => (
  <div>
    {props.forms && props.forms[0].id}
    <p className="foo">Foo: {props.foo}</p>
    <p className="baz">Baz: {props.baz}</p>
  </div>
);

const mapStateToProps = state => ({
  forms: getFormsFromState(state)
});

function mapDispatchToProps(dispatch) {
  return {
    UshLogoActions: bindActionCreators(UshLogoActions, dispatch)
  };
}

MyComponent.propTypes = {
  foo: PropTypes.number.isRequired,
  baz: PropTypes.number.isRequired,
  forms: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  ).isRequired
};

export { MyComponent as PlainUshLogo };

export default connectWithStore(
  MyComponent,
  mapStateToProps,
  mapDispatchToProps
);
