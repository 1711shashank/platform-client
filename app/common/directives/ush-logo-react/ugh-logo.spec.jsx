import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { PlainUshLogo as MyComponent } from "./ush-logo";

test("foo and bar match the props", () => {
  const forms = [{ id: "junk" }];
  const actions = {
  	receiveForms: jest.fn()
  };
  // Note that the way you pass down props in react vs. angular is different
  // Therefore you need pass down props differently
  const component = renderer.create(
    <MyComponent forms={forms} foo={1} baz={2} UshLogoActions={actions} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("MyComponent to correctly render foo and baz in p tags", () => {
  const forms = [{ id: "junk" }];
  const actions = {
  	receiveForms: jest.fn()
  };
  // Render a checkbox with label in the document
  const fooBaz = shallow(<MyComponent forms={forms} foo={4} baz={1} UshLogoActions={actions} />);

  expect(fooBaz.find("p").length).toBe(2);
  expect(fooBaz.find("p.foo").text()).toBe("Foo: 4");
  expect(fooBaz.find("p.baz").text()).toBe("Baz: 1");
});
