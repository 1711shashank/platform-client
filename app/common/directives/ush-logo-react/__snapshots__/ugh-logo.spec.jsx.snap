// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foo and bar match the props 1`] = `
<div>
  junk
  <p
    className="foo"
  >
    Foo: 
    1
  </p>
  <p
    className="baz"
  >
    Baz: 
    2
  </p>
</div>
`;
